### ISSUE TITLE
Develop and Implement `train_model()`

### ISSUE THREAD

**Issue by jciglesias** [jciglesias/total_perspective_vortex#2]
#### Current State:
- A placeholder function is implemented to train models, but there is no real training logic.

#### Requirements (from the subject):
- Train a machine learning model to **classify EEG epochs** based on motor imagery tasks (e.g., hand vs. foot movements).
- Use **scikit-learn pipelines** to integrate preprocessing (e.g., wavelet transforms) and machine learning.

#### Suggested Steps:
1. **Train-Test-Split**:
    - **Split Data**: Split epochs data into training, validation, and testing subsets.
    - Example (assuming `epochs` corresponds to epoched EEG data):
``` python
from sklearn.model_selection import train_test_split

X = epochs.get_data()  # EEG data (num_epochs, num_channels, num_samples)
y = epochs.events[:, -1]  # Labels from events
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
```
1. **Build Classification Pipeline**:
    - Implement a **scikit-learn pipeline**, as requested by the subject, which includes:
        - **Wavelet Transformation** (via `pywt`).
        - **Flattening data** for machine learning models (reshape 2D arrays).
        - **Dimensionality Reduction** (optional, e.g., PCA).
        - **Classifier** (e.g., Random Forest, SVM).

Example Pipeline:
``` python
from sklearn.pipeline import Pipeline
from sklearn.ensemble import RandomForestClassifier
from sklearn.preprocessing import StandardScaler
import numpy as np


# Add Wavelet Transform step
class WaveletTransform:
    def fit(self, X, y=None):
        return self

    def transform(self, X):
        wavelet = 'db4'
        transformed_data = [pywt.dwt(x, wavelet)[0] for x in X]
        return np.array(transformed_data)


pipeline = Pipeline([
    ('wavelet', WaveletTransform()),  # Apply wavelet transform
    ('scaler', StandardScaler()),  # Standardize features
    ('classifier', RandomForestClassifier())  # Train a classifier
])
pipeline.fit(X_train, y_train)
```
1. **Save Pipeline in Session State**:
    - Save the trained pipeline for predictions:
``` python
pipeline.fit(X_train, y_train)
st.session_state[f"{subject}_{task}_pipeline"] = pipeline
```
1. **Evaluate Model Performance**:
    - Compute metrics like accuracy, precision, recall on the validation/test data:
``` python
from sklearn.metrics import accuracy_score, classification_report

y_pred = pipeline.predict(X_test)
st.write("Accuracy:", accuracy_score(y_test, y_pred))
st.text(classification_report(y_test, y_pred))
```
